backend/services/transcoder/transcoder.go

```diff
@@ -14,34 +14,63 @@ import (
 	"webserver/models"
 	"webserver/services"
 
+	"github.com/alitto/pond"
 	cmap "github.com/orcaman/concurrent-map/v2"
 	log "github.com/sirupsen/logrus"
-	"github.com/sourcegraph/conc/pool"
 	"github.com/volatiletech/sqlboiler/v4/boil"
 )
 
 // A go package that implements a worker pool to process files in minio using ffmpeg into mpeg-dash format
 // and stores the output in minio.
 type transcoder struct {
 	*services.Group
-	pool *pool.Pool
+	pool *pond.WorkerPool
 
 	progress cmap.ConcurrentMap[int64, int]
+	started  bool
 }
 
-func New(grp *services.Group, workers int) *transcoder {
-	return &transcoder{
-		pool:  pool.New().WithMaxGoroutines(workers),
+func New(grp *services.Group, workers int) (services.Transcoder, error) {
+	t := &transcoder{
+		pool:  pond.New(workers, 1000),
 		Group: grp,
 		progress: cmap.NewWithCustomShardingFunction[int64, int](func(key int64) uint32 {
 			// Copilot recommended this i have no idea if its correct
 			return uint32(key % 10)
 		}),
+		started: false,
 	}
+
+	// Find all clips that are marked as processing while starting to resume their processing
+	orphanedClips, err := t.Clips.FindMany(context.Background(), models.ClipWhere.Processing.EQ(true))
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, clip := range orphanedClips {
+		if err := t.Queue(context.Background(), clip); err != nil {
+			return nil, err
+		}
+	}
+
+	return t, nil
+}
+
+func (t *transcoder) Start() error {
+	if t.started {
+		return fmt.Errorf("transcoder already started")
```

Comment:
[nit] Should we move these to constants instead of having "magic" strings?

Reply:
Agree, copilot wrote this for me lmao